services/github/github-package-lock-json.service.js

```diff
@@ -0,0 +1,126 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const {
+  isPackageLockJsonWithDependencies,
+  getLockDependencyVersion,
+} = require('../package-lock-json-helpers')
+const { fetchJsonFromRepo } = require('./github-common-fetch')
+const { documentation } = require('./github-helpers')
+const { BaseJsonService } = require('..')
+
+const keywords = ['npm', 'node']
+
+const dependencyQueryParamSchema = Joi.object({
+  filename: Joi.string(),
+}).required()
+
+class GithubPackageLockJsonDependencyVersion extends BaseJsonService {
+  static get category() {
+    return 'platform-support'
+  }
+
+  static get route() {
+    return {
+      base: 'github/package-lock-json/dependency-version',
+      pattern: ':user/:repo/:scope(@[^/]+)?/:packageName/:branch*',
+      queryParamSchema: dependencyQueryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub package-lock.json dependency version',
+        pattern: ':user/:repo/:packageName',
+        namedParams: {
+          user: 'developit',
+          repo: 'microbundle',
+          packageName: 'rollup',
+        },
+        staticPreview: this.render({
+          dependency: 'rollup',
+          range: '1.29.1',
+        }),
+        documentation,
+        keywords,
+      },
+      {
+        title: 'GitHub package-lock.json dependency version (on branch)',
+        pattern: ':user/:repo/dev/:scope?/:packageName/:branch*',
+        namedParams: {
+          user: 'octokit',
+          repo: 'rest.js',
+          branch: 'master',
+          scope: '@babel',
+          packageName: 'core',
+        },
+        staticPreview: this.render({
+          dependency: '@babel/core',
+          range: '7.8.4',
+        }),
+        documentation,
+        keywords,
+      },
+      {
+        title:
+          'GitHub package-lock.json dependency version (subfolder of monorepo)',
+        pattern: ':user/:repo/:packageName',
+        namedParams: {
+          user: 'OriginProtocol',
+          repo: 'origin',
+          packageName: 'davidshimjs-qrcodejs',
+        },
+        queryParams: {
+          filename: 'origin/dapps/marketplace/package-lock.json',
```

Comment:
the path to the file shouldn't include a repo/remote/branch name, so we should remove `origin` from here so the live example values presented in the badge builder window will work

```suggestion
          filename: 'dapps/marketplace/package-lock.json',
```

https://shields-staging-pr-4722.herokuapp.com/github/package-lock-json/dependency-version/OriginProtocol/origin/davidshimjs-qrcodejs?filename=origin%2Fdapps%2Fmarketplace%2Fpackage-lock.json
![](https://shields-staging-pr-4722.herokuapp.com/github/package-lock-json/dependency-version/OriginProtocol/origin/davidshimjs-qrcodejs?filename=origin%2Fdapps%2Fmarketplace%2Fpackage-lock.json)

https://shields-staging-pr-4722.herokuapp.com/github/package-lock-json/dependency-version/OriginProtocol/origin/davidshimjs-qrcodejs?filename=dapps%2Fmarketplace%2Fpackage-lock.json
![](https://shields-staging-pr-4722.herokuapp.com/github/package-lock-json/dependency-version/OriginProtocol/origin/davidshimjs-qrcodejs?filename=dapps%2Fmarketplace%2Fpackage-lock.json)